drivers/lora/shell.c

```diff
@@ -71,21 +71,21 @@ static int parse_long_range(long *out, const struct shell *shell,
 static int parse_freq(uint32_t *out, const struct shell *shell, const char *arg)
 {
 	char *eptr;
-	long long llval;
+	unsigned long val;
 
-	llval = strtoll(arg, &eptr, 0);
+	val = strtoul(arg, &eptr, 0);
 	if (*eptr != '\0') {
 		shell_error(shell, "Invalid frequency, '%s' is not an integer",
 			    arg);
 		return -EINVAL;
 	}
 
-	if (llval < 0 || llval > UINT32_MAX) {
-		shell_error(shell, "Frequency %lli out of range", llval);
+	if (val > UINT32_MAX) {
```

Comment:
I don't believe this is correct.  The return value of `strtoul` if the value would exceed `ULONG_MAX` (i.e. not fit in a 32-bit unsigned) is `ULONG_MAX` which is probably `UINT32_MAX` so the test would never succeed.  Better to error if the returned value is `ULONG_MAX` (not `UINT32_MAX`).

If `UINT32_MAX` is an actual valid value that should be accepted, rather than one that could be interpreted solely as an overflow, `errno` should be cleared before the call and checked after the call.

Reply:
Good catch! I've updated commit to check for `ULONG_MAX` as you proposed and now it works properly for overflows :)